src/main.c

```diff
@@ -3440,17 +3440,36 @@ int main(int argc, char **argv)
         conf.error_log = NULL;
     }
 
-    { /* raise RLIMIT_NOFILE */
-        struct rlimit limit;
+    {
+        struct rlimit limit = {0};
+
         if (getrlimit(RLIMIT_NOFILE, &limit) == 0) {
+            if (conf.max_connections > (int)limit.rlim_max) {
+                fprintf(stderr, "[error] The 'max-connections'=[%d] configuration value "
+                                "should not exceed the file descriptor limit of "
+                                "the process 'RLIMIT_NOFILE'=[%lu]\n",
+                        conf.max_connections, limit.rlim_max);
```

Comment:
Would you mind letting clang-format fold the lines? You can run `clang-format -i src/main.c` to format a particular source file, or run `make -f misc/regen.mk clang-format` to clang-format the entire source tree.

Unless there is a strong reason, we do not manually fold the source code. Instead, we let clang-format do the job because we do not want to discuss about how the code should be indented.

Same argument applies to other invocations of fprintf.

Reply:
Thanks for your comments.  I have incorporated them in [ad08259b](https://github.com/h2o/h2o/pull/2705/commits/ad08259bdb22f17a1b015725bae40efc0861fa00).